content/3_Silver/Sorting_Custom.mdx

```diff
@@ -815,19 +815,62 @@ public class RectangularPasture {
 				                 Math.max(ys[j], ys[k]), n);
 			}
 		}
+
 		System.out.println(answer);
 	}
 }
-// EndCodeSnip
 ```
 
-The solution uses a
-[lambda](https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html)
-function as the custom comparator, which our guide didn't discuss, but it should
-be apparent which coordinate (`x` or `y`) that the comparator is sorting by.
-
 </JavaSection>
+<PySection>
+
+<Warning>
+
+Due to Python's constant factor, the below code will TLE
+on a couple test cases despite having the correct time complexity.
+
+</Warning>
+
+```py
+import sys
+
+input = sys.stdin.readline
+
+n = int(input().strip())
+points = [list(map(int, input().strip().split())) for _ in range(n)]
 
+points.sort(key=lambda i: i[1])
+for i in range(n):
+	points[i][1] = i + 1
+
+points.sort(key=lambda i: i[0])
+for i in range(n):
+	points[i][0] = i + 1
+
+psa = [[0 for _ in range(n + 1)] for _ in range(n + 1)]
```

Comment:
ik this is prefix sums but what does `psa` actually stand for

Reply:
Prefix Sum Array
I know it's a 2d psa, but i think it gets the idea across